### PR TITLE
Improve project setup

### LINKED ISSUE #3: Add release process tooling

### LINKED ISSUE #14: Add airbase as parent
For release tooling and much more.

### PR DESCRIPTION
**Pull request by mosabua**
- Add airbase as parent to enable release in future
- Add necessary details for release
- Add airbase property overrides to skip various checks to allow passing build
- Move test code into correct path
- Centralize property declarations in parent pom
- Improve dependency declarations and add version overrides as minimally needed

Fixes #3 
Fixes #14

### CODE REVIEW

**Comment by mosabua** [comment]
Currently there are test failures related to some classpath issues .. @willmostly and myself are looking but if anybody else wants to chime in that would help

File: baseapp/pom.xml [hunk 1]
```diff
@@ -38,6 +30,10 @@
             <artifactId>guice</artifactId>
             <version>${guice.version}</version>
         </dependency>
+        <dependency>
+          <groupId>com.fasterxml.jackson.core</groupId>
+          <artifactId>jackson-core</artifactId>
```
Comment (dain) [inline baseapp/pom.xml:35]: This should be indented 4 spaces

File: baseapp/pom.xml [hunk 2]
```diff
@@ -85,6 +81,12 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+              <groupId>org.apache.maven.plugins</groupId>              <artifactId>maven-surefire-plugin</artifactId>
```
Comment (dain) [inline baseapp/pom.xml:85]: formatting is off here

File: pom.xml [hunk 3]
```diff
@@ -5,28 +5,79 @@
...
+        <license>
+            <name>Apache License 2.0</name>
+            <url>http://www.apache.org/licenses/LICENSE-2.0</url>
+            <distribution>repo</distribution>
+        </license>
+    </licenses>
+    <developers>
+      <developer>
+        <name>Trino Gateway contributors</name>
+        <url>https://github.com/trinodb/trino-gateway/graphs/contributors</url>
```
Comment (dain) [inline pom.xml:34]: 4 spaces for this file also

**Review by dain — COMMENTED** [review]
There are some minor formatting issues

File: baseapp/pom.xml (hunk 1 above)
Comment (mosabua) [reply baseapp/pom.xml:35]: Let me check them all

File: pom.xml (hunk 3 above)
Comment (mosabua) [reply pom.xml:34]: done .. so sorry .. four is just sooo wide

File: pom.xml [hunk 4]
```diff
@@ -99,45 +150,4 @@
         <module>gateway-ha</module>
     </modules>
 
-    <build>
-        <plugins>
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
```
Comment (mosabua) [inline pom.xml:105]: This is removed because we want to move to the Trino style anyway

File: pom.xml [hunk 5]
```diff
@@ -5,28 +5,79 @@
...
+    </parent>
+
     <groupId>io.trino.gateway</groupId>
     <artifactId>trinogateway-parent</artifactId>
-    <name>trinogateway-parent</name>
     <packaging>pom</packaging>
     <version>3-SNAPSHOT</version>
 
+    <name>trinogateway-parent</name>
+    <description>A smart load balancer and query route for Trino</description>
```
Comment (willmostly) [inline pom.xml:20]: route => router

**Review by willmostly — APPROVED** [review]
one nit, otherwise LGTM

File: pom.xml (hunk 5 above)
Comment (mosabua) [reply pom.xml]: Fixed

**Comment by wendigo** [comment]
@mosabua can you split this PR into multiple commits?

File: gateway-ha/pom.xml [hunk 6]
```diff
@@ -140,7 +128,7 @@
         <dependency>
             <groupId>mysql</groupId>
             <artifactId>mysql-connector-java</artifactId>
-            <version>8.0.17</version>
+            <version>${mysqlconnector.version}</version>
```
Comment (wendigo) [inline gateway-ha/pom.xml:131]: nit: dep.mysql.version

File: pom.xml [hunk 7]
```diff
@@ -5,28 +5,79 @@
...
+        <dep.jackson.version>2.10.1</dep.jackson.version>
+        <dep.logback.version>1.2.12</dep.logback.version>
+        <dep.slf4j.version>${slf4j.version}</dep.slf4j.version>
+
+        <!-- dependency versions -->
+        <activejdbc.version>2.3</activejdbc.version>
+        <dropwizard.version>1.3.29</dropwizard.version>
+        <ehcache.version>3.8.1</ehcache.version>
+        <guava.version>23.0</guava.version>
+        <guice.version>4.1.0</guice.version>
```
Comment (wendigo) [inline pom.xml:72]: why 4.1.0? Trino uses 7.0.0

File: pom.xml [hunk 8]
```diff
@@ -5,28 +5,79 @@
...
+        <air.check.skip-dependency>true</air.check.skip-dependency>
+        <air.check.skip-duplicate-finder>true</air.check.skip-duplicate-finder>
+        <air.check.skip-modernizer>true</air.check.skip-modernizer>
+        <air.check.fail-spotbugs>false</air.check.fail-spotbugs>
+        <air.check.skip-pmd>true</air.check.skip-pmd>
+
+        <!-- overriding version from airbase -->
+        <dep.jackson.version>2.10.1</dep.jackson.version>
+        <dep.logback.version>1.2.12</dep.logback.version>
+        <dep.slf4j.version>${slf4j.version}</dep.slf4j.version>
```
Comment (wendigo) [inline pom.xml:65]: it's declared by airbase. drop

File: pom.xml [hunk 9]
```diff
@@ -5,28 +5,79 @@
...
+        <air.check.skip-enforcer>true</air.check.skip-enforcer>
+        <air.check.skip-license>true</air.check.skip-license>
+        <air.check.skip-dependency>true</air.check.skip-dependency>
+        <air.check.skip-duplicate-finder>true</air.check.skip-duplicate-finder>
+        <air.check.skip-modernizer>true</air.check.skip-modernizer>
+        <air.check.fail-spotbugs>false</air.check.fail-spotbugs>
+        <air.check.skip-pmd>true</air.check.skip-pmd>
+
+        <!-- overriding version from airbase -->
+        <dep.jackson.version>2.10.1</dep.jackson.version>
```
Comment (wendigo) [inline pom.xml:63]: it's declared by airbase. drop

File: pom.xml [hunk 10]
```diff
@@ -5,28 +5,79 @@
...
+        <dep.logback.version>1.2.12</dep.logback.version>
```
Comment (wendigo) [inline pom.xml:64]: it's declared by airbase. drop

Comment (mosabua) [reply pom.xml:64]: This is a necessary override .. without it the project does not work/build ..we will update deps after first release

File: pom.xml (hunk 9 above)
Comment (mosabua) [reply pom.xml:63]: Necessary override

File: pom.xml [hunk 11]
```diff
@@ -5,28 +5,79 @@
...
+        <dep.slf4j.version>${slf4j.version}</dep.slf4j.version>
+
+        <!-- dependency versions -->
+        <activejdbc.version>2.3</activejdbc.version>
+        <dropwizard.version>1.3.29</dropwizard.version>
+        <ehcache.version>3.8.1</ehcache.version>
+        <guava.version>23.0</guava.version>
```
[the remark this answers is not in the document]
Comment (mosabua) [reply pom.xml:71]: we will work on updating all dependencies .. after this PR and first release .. there is a LOT to update

File: pom.xml (hunk 7 above)
Comment (mosabua) [reply pom.xml:72]: Because that is what the current codebase uses and needs.

File: gateway-ha/pom.xml (hunk 6 above)
Comment (mosabua) [reply gateway-ha/pom.xml:131]: we will clean this up separately later .. this PR is minimal

**Comment by mosabua** [comment]
> @mosabua can you split this PR into multiple commits?

Not really .. this current setup is necessary for it to build. If I split it up the commits in between wont build.

**Comment by andythsu** [comment]
I have a script running some tasks against the gateway with changes in this PR and it seems to be running fine. It's good for me.

**Review by andythsu — APPROVED** [review]

**Comment by mosabua** [comment]
Rebase #17 from main into here now .. everything still working.

**Review by vishalya — APPROVED** [review]
It's in a decent shape now. I am able to compile and test locally.

**Review by dain — APPROVED** [review]

**Pull request merged**